guides/building-helm-charts-for-spk.md

```diff
@@ -0,0 +1,257 @@
+which is written to `bedrock.yaml`. The relevant lines from `bedrock.yaml` are
+below:
+
+```yaml
+helm:
+  chart:
+    branch: master
+    git: "https://dev.azure.com/fabrikam/frontend/_git/charts"
+    path: frontend
+```
+
+The above configuration presumes that you have a _seperate_ git repository that
+contains a helm chart in the `frontend` path. This helm chart is further
+configured by a scaffolded configuration from the `hld-lifecycle` pipeline. For
+reference, a "compatible" helm chart for `spk` is
+[provided here](./sample-helm-chart).
+
+#### Ring level configuration
+
+![Generated ring configuration common.yaml](./images/spk-hld-ring-configuration.png)
+
+The above `common.yaml` configures the subcomponent `chart` from the
+`component.yaml` above to configure a `serviceName` for the helm chart,
+`fabrikam-k8s-svc-dev`. This configuration is produced from the `hld-lifecycle`
+pipeline, and consumes configuration a user provides via `spk service create`,
+which is written to `bedrock.yaml`. The relevant lines consumed from
+`bedrock.yaml` are below:
+
+```yaml
+rings:
+  dev:
+    isDefault: true
+services:
+  ./
+    ...
+    k8sBackend: 'fabrikam-k8s-svc'
+```
+
+The `serviceName` is generated from a combination of the `k8sBacend`,
+`fabrikam-k8s-svc`, and the ring name, in this case `dev`.
+
+#### Static configuration
+
+The second subcomponent in the `component.yaml` above references a subdirectory
+at the same level, named `static`. This subdirectory contains a generated
+Traefik IngressRoute from the `hld-lifecycle` pipeline, which is a static
+Kubernetes manifest and can be directly applied to a cluster.
+
+![Generated Ingress Route](./images/spk-hld-generated-ingress-route.png)
+
+The `hld-lifecycle` pipeline scaffolds the above
+[Traefik2 IngressRoute](https://docs.traefik.io/v2.0/providers/kubernetes-crd/)
+from configuration a user provides via `spk service create`, which is writen to
+`bedrock.yaml`. The relevant lines consumed from `bedrock.yaml` are below:
+
+```yaml
+rings:
+  dev:
+    isDefault: true
+services:
+  ./
+    ...
+    displayName: 'fabrikam'
+    k8sBackend: 'fabrikam-k8s-svc'
+    pathPrefix: 'fabrikam-service'
+    pathPrefixMajorVersion: 'v1'
+```
+
+- The name of the IngressRoute, `fabrikam-dev` is generated from a combination
+  of the displayName and the ring name (in this case, `dev`).
+
+- The match rule, `PathPrefix('/v1/fabrikam-service') && Headers('Ring', 'dev')`
+  is generated from a combination of the `pathPrefixMajorVersion`, the
+  `pathPrefix` and the ring name (once again `dev`).
+
+- The backing service `fabrikam-k8s-svc-dev`, (highlighted in green) is
+  generated from `k8sBackend` and the ring name (`dev`).
+
+#### Helm Configuration
+
+Take a look at the [packaged reference helm chart](./sample-helm-chart), it
+contains a `values.yaml`, which is a set of default values for helm charts:
+
+```yaml
+image:
+  tag: stable
+
```

Comment:
do we need `repository` here?

Reply:
Yes, that's now expected: https://github.com/CatalystCode/spk/pull/341

Reply:
will update this